flink-table/flink-sql-parser/src/main/java/org/apache/flink/sql/parser/ddl/SqlCreateTableAs.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.sql.parser.ddl;
+
+import org.apache.flink.sql.parser.ExtendedSqlNode;
+import org.apache.flink.sql.parser.ddl.constraint.SqlTableConstraint;
+import org.apache.flink.sql.parser.error.SqlValidateException;
+
+import org.apache.calcite.sql.SqlCreate;
+import org.apache.calcite.sql.SqlKind;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.SqlOperator;
+import org.apache.calcite.sql.SqlSpecialOperator;
+import org.apache.calcite.sql.SqlWriter;
+import org.apache.calcite.util.ImmutableNullableList;
+
+import javax.annotation.Nonnull;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+/** CREATE TABLE AS SELECT DDL sql call. */
+public class SqlCreateTableAs extends SqlCreate implements ExtendedSqlNode {
+
+    public static final SqlSpecialOperator OPERATOR =
+            new SqlSpecialOperator("CREATE TABLE AS SELECT", SqlKind.OTHER_DDL);
+
+    private final SqlCreateTable sqlCreateTable;
+    private final SqlNode asQuery;
+
+    public SqlCreateTableAs(SqlCreateTable sqlCreateTable, SqlNode asQuery) {
+        super(
+                OPERATOR,
+                sqlCreateTable.getParserPosition(),
+                sqlCreateTable.getReplace(),
+                sqlCreateTable.isIfNotExists());
+        this.sqlCreateTable = sqlCreateTable;
```

Comment:
It's better to check the input sqlCreateTable does not have tableLike.

Reply:
SQL syntax has defined that `AS` and `LIKE` can only appear one, theoretically `SqlCreateTableAs` will not appear `SqlCreateTable#tableLike` is not null.
see UT: `FlinkSqlParserImplTest#testCreateTableAsSelectWithCreateTableLike`

Reply:
Since Like and AS are all optional clauses for a create table query, it's wierd that `LIKE` is a property of `SqlCreateTable`, but `AS` is a property of subclass.
It's better to introduce `SqlCreateTable` as base class. And introduce `SqlCreateTableLike` and `SqlCreateTableAs` to handle `LIKE` and `AS` clauses.

Reply:
@beyond1920  @lsyldliu  thanks for your advice. I will try to introduce `SqlCreateTableLike` and `SqlCreateTableAs`